gradio/helpers.py

```diff
@@ -259,16 +259,14 @@ def create(self) -> None:
 
         async def load_example(example_id):
             processed_example = self.non_none_processed_examples[example_id]
-            examples = utils.resolve_singleton(processed_example)
-
-            return (
-                update(value=examples, **self.dataset.component_props[0])
-                if not isinstance(examples, list)
-                else [
-                    update(value=ex, **self.dataset.component_props[i])
-                    for i, ex in enumerate(examples)
-                ]
-            )
+            if len(self.inputs_with_examples) == 1:
+                return update(
+                    value=processed_example[0], **self.dataset.component_props[0]
+                )
+            return [
+                update(value=processed_example[i], **self.dataset.component_props[i])
+                for i in range(len(self.inputs_with_examples))
```

Comment:
Should be `self.inputs_with_examples` not `examples`

Reply:
Ah cool. I have to admit all of this code confused me a great deal. I still don't know what all of the references to different examples/ inputs are.

Reply:
Yeah the code for examples is messy, will clean up at some point. Your PR using gr.update is nice though, captures an edge case we didn’t think about originally!